internal/k8s/controller.go

```diff
@@ -1408,22 +1408,25 @@ func getStatusFromEventTitle(eventTitle string) string {
 		return conf_v1.StateWarning
 	case "AddedOrUpdated", "Updated":
 		return conf_v1.StateValid
-	default:
-		return "Unknown"
 	}
 }
 
-func (lbc *LoadBalancerController) updateVirtualServerVirtualServerRouteStatusesFromEvents() error {
-	vss := lbc.getVirtualServers()
+func (lbc *LoadBalancerController) updateVirtualServersStatusFromEvents() error {
 	var allErrs []error
-	for _, vs := range vss {
+	for _, obj := range lbc.virtualServerLister.List() {
+		vs := obj.(*conf_v1.VirtualServer)
+
 		events, err := lbc.client.CoreV1().Events(vs.Namespace).List(context.TODO(),
 			meta_v1.ListOptions{FieldSelector: fmt.Sprintf("involvedObject.name=%v,involvedObject.uid=%v", vs.Name, vs.UID)})
 		if err != nil {
 			allErrs = append(allErrs, fmt.Errorf("error trying to get events for VirtualServer %v/%v: %v", vs.Namespace, vs.Name, err))
 			break
 		}
 
+		if len(events.Items) == 0 {
+			break
```

Comment:
break will stop processing events for all resources if one has zero events. shall we use continue here?

Reply:
Continue yeah, my mistake.